launcher/sdw_updater_gui/UpdaterApp.py

```diff
@@ -0,0 +1,256 @@
+
+    def __init__(self):
+        QThread.__init__(self)
+
+    def run(self):
+        results = {}
+
+        for vm in VMS:
+            progress = {
+                "output": "Checking for updates for {}".format(vm),
+                "run_completed": False,
+            }
+            self.signal.emit(progress)
+            try:
+                p = subprocess.Popen(
+                    ["securedrop-updater-cli", "--check", vm],
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                )
+                out, err = p.communicate()
+                if p.returncode == 0:
+                    results[vm] = False
+                else:
+                    results[vm] = True
+                progress = {"output": str(out), "run_completed": False}
+                self.signal.emit(progress)
+            except subprocess.CalledProcessError as e:
+                results[vm] = True
+                progress = {"output": str(e), "run_completed": False}
+                self.signal.emit(progress)
+
+        result = {
+            "run_completed": True,
+            "action": "update",
+            "dom0_update_required": results["dom0"],
+            "sd-svs-disp_update_required": results["sd-svs-disp"],
+            "fedora_update_required": results["fedora"],
+            "output": "Update check complete.",
+        }
+        self.signal.emit(result)
+
+
+# This thread will upgrade VMs that require upgrades
+class UpgradeThread(QThread):
+    signal = pyqtSignal("PyQt_PyObject")
+    vms_to_update = []
+
+    def __init__(self, vms):
+        QThread.__init__(self)
+        self.vms_to_update = vms
+
+    def run(self):
+        results = {}
+        for vm in self.vms_to_update:
+            progress = {
+                "output": "Applying updates for {}".format(vm),
+                "run_completed": False,
+            }
+            self.signal.emit(progress)
+
+            try:
+                p = subprocess.Popen(
+                    ["securedrop-updater-cli", "--upgrade", vm],
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                )
+                out, err = p.communicate()
+                if p.returncode == 0:
+                    results[vm] = False
+                else:
+                    results[vm] = True
+                progress = {"output": str(out), "run_completed": False}
+                self.signal.emit(progress)
+            except subprocess.CalledProcessError as e:
+                results[vm] = True
+                progress = {"output": str(e), "run_completed": False}
+                self.signal.emit(progress)
+
+        result = {
+            "run_completed": True,
+            "action": "upgrade",
+            "dom0_upgrade_success": results.get("dom0", True),
+            "sd-svs-disp_upgrade_success": results.get("sd-svs-disp", True),
+            "fedora_upgrade_success": results.get("fedora", True),
```

Comment:
why `results.get("fedora", True)` here? do we want to use a default value of True? (if this is indeed what you want, let's add an explanatory comment)